vitrina/requests/views.py

```diff
@@ -14,3 +17,47 @@ class RequestListView(ListView):
 class RequestDetailView(DetailView):
     model = Request
     template_name = 'vitrina/requests/detail.html'
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        request = context_data.get('request')
+
+        structure_data = request.structure_data.split(";") if request.structure_data else []
+        structure = []
+        for struct in structure_data:
+            data = struct.split(",")
+            if data:
+                structure.append({
+                    "data_title": data[0],
+                    "dictionary_title": data[1],
+                    "data_type": data[2],
+                    "data_notes": data[3],
+                })
+
+        dataset = None
+        if request.dataset_id:
+            try:
+                dataset = Dataset.public.get(pk=request.dataset_id)
+            except ObjectDoesNotExist:
+                pass
+
+        liked = False
+        if self.request.user and self.request.user.pk:
+            user_like = UserLike.objects.filter(request_id=request.pk, user_id=self.request.user.pk)
```

Comment:
```suggestion
            user_like = UserLike.objects.filter(request=request, user=self.request.user)
```

Reply:
UserLike doesn't have fields request and user, would need to change model.

Reply:
Ok, lets leave it like that, user likes should be converted to generic model anyway.